ner_v2/detectors/numeral/number/standard_number_detector.py

```diff
@@ -65,104 +71,158 @@ def init_regex_and_parser(self, data_directory_path):
         Returns:
             None
         """
-        data_df = pd.read_csv(os.path.join(data_directory_path, NUMBER_DATA_CONSTANT_FILE), encoding='utf-8')
-        for index, row in data_df.iterrows():
-            number = row[NUMBER_DATA_FILE_NUMBER]
-            numerals = row[NUMBER_DATA_FILE_NUMERALS]
+        # create number_words dict having number variants and their corresponding scale and increment value
+        # create language_scale_map dict having scale variants and their corresponding value
+        numeral_df = pd.read_csv(os.path.join(data_directory_path, NUMBER_NUMERAL_CONSTANT_FILE), encoding='utf-8')
+        for index, row in numeral_df.iterrows():
+            name_variants = self._get_list_from_pipe_sep_string(row[NUMBER_DATA_FILE_NAME_VARIANTS])
             value = row[NUMBER_DATA_FILE_VALUE]
             if float(value).is_integer():
                 value = int(row[NUMBER_DATA_FILE_VALUE])
             number_type = row[NUMBER_DATA_FILE_TYPE]
 
-            if value in NUMBER_DIGIT_UNITS:
-                self.language_number_map[number] = value
-                self.language_number_map[str(value)] = value
-
-            numerals_list = self._get_numerals_list(numerals)
             if number_type == NUMBER_TYPE_UNIT:
-                self.numbers_word[number] = (1, value)
-                self.numbers_word[str(value)] = (1, value)
-                for numeral in numerals_list:
+                for numeral in name_variants:
                     self.numbers_word[numeral] = (1, value)
```

Comment:
A better way would have been to use `collections.namedtuple`
`NumberVariant = collections.namedtuple('NumberVariant', ['scale', 'value'])`

Then you can make tuples like
`NumberVariant(scale=1, value=value)`

This avoids the hassle of using indices.

But this change would need changes wherever these tuples are being used

Reply:
Done